externals/kyuubi-spark-sql-engine/pom.xml

```diff
@@ -51,6 +51,12 @@
             <version>${project.version}</version>
         </dependency>
 
+        <dependency>
```

Comment:
any idea about removing this from the engine？ It makes engine module hard for long-term maintenance and cross version support

Reply:
@yaooqinn @ulysses-you @cfmcgrady 
https://github.com/apache/incubator-kyuubi/pull/3537#issuecomment-1260305095 Here just discussed the dependency with the plugin. @wForget suggested extracting the common module to reuse `SQLLineageParserHelper`, can you give some other advice?

Reply:
the lineage module does not shade the spark, so I think the worst case is the lineage can not work and the engine won't be broken ?

we can add try-catch for those code in engine and narrow the exception

Reply:
A spark engine plugin can depend on engine lib(not suggested, but ok) and spark libs for the compilation phase, but the engine lib can not do the opposite.

Reply:
how about have a rule or logic inside the lineage plugin, the engine side can push down a config or something, then you determine the output

Reply:
@yaooqinn Can we use `optional` dependencies here? Refer to the definition of maven `optional`: 

> The idea is that some of the dependencies are only used for certain features in the project and will not be needed if that feature isn't used

Reply:
Can we just make the full functionality on the plugin side? IMHO, it is not necessary to include a dependency based on my experience to create more than one plugins

Reply:
@yaooqinn I have removed the dependency with your advice